src/openapi-generator.ts

```diff
@@ -620,6 +640,40 @@ export class OpenAPIGenerator {
     };
   }
 
+  private openApiVersionSatisfies = (
+    inputVersion: OpenApiVersion,
+    comparison: OpenApiVersion
+  ): boolean =>
+    openApiVersions.indexOf(inputVersion) >=
+    openApiVersions.indexOf(comparison);
+
+  private mapNullableType(
+    type: SchemaObject['type'],
+    isNullable: boolean
+  ): Pick<SchemaObject, 'type' | 'nullable'> {
+    // Open API 3.1.0 made the `nullable` key invalid and instead you use type arrays
+    if (this.openApiVersionSatisfies(this.openAPIVersion, '3.1.0')) {
+      if (isNullable) {
+        if (type === undefined) {
```

Comment:
Hmm this seems a little but obscure. Considering the usage is `...(isNullable ? [this.mapNullableType(undefined, isNullable)] : [])`. Basically we always call this function with `undefined` and `true`. Maybe it would be better to create a separate function for thosecases. It can either handle the whole ternary operator as well => returns an array itself. Or it can just handle "the "generation of the `nullable` parameter for oneOf/allOf/anyOf"